crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -2596,7 +2605,16 @@ fn build_int_unary_op<'a, 'ctx, 'env>(
                         }
                     }
                     PtrWidth::Bytes8 => {
-                        if target_int_width.stack_size() as usize > env.target_info.ptr_size() {
+                        let return_by_pointer = {
+                            if env.target_info.operating_system
+                                == roc_target::OperatingSystem::Windows
+                            {
+                                target_int_width.stack_size() as usize >= env.target_info.ptr_size()
+                            } else {
+                                target_int_width.stack_size() as usize > env.target_info.ptr_size()
```

Comment:
Feels weird that these are different. Are we sure that this change shouldn't be for all OSes?

Reply:
Yes, a number of tests will fail without this. @folkertdev can probably explain the details, they wen't a little over my head at the time.